app/package/controllers.py

```diff
@@ -440,12 +441,14 @@ def get_metadata(publisher, package):
             return handle_error('DATA_NOT_FOUND',
                                 'No metadata found for the package',
                                 404)
+        if not isinstance(data.descriptor, dict):
+            raise Exception("descriptor should be type of dict")
```

Comment:
@Fak3 There were places where we were putting as a string and db was not throwing any error. I changed the places where string was getting written in db.